spec/services/doc_auth/mock/doc_auth_mock_client_spec.rb

```diff
@@ -185,27 +185,47 @@
   end
 
   describe 'selfie check performed flag' do
-    let(:response) do
-      client.post_images(
-        front_image: DocAuthImageFixtures.document_front_image,
-        back_image: DocAuthImageFixtures.document_back_image,
-        liveness_checking_required: liveness_checking_required,
-      )
-    end
-
     context 'when a liveness check is required' do
       let(:liveness_checking_required) { true }
 
+      image = <<~YAML
+        portrait_match_results:
+          face_match_result: Pass
+          face_error_message: 'Successful. Liveness: Live'
+        doc_auth_result: Passed
+        failed_alerts: []
+      YAML
+
       it 'sets selfie_check_performed to true' do
+        response = client.post_images(
+          front_image: image,
+          back_image: image,
+          liveness_checking_required: liveness_checking_required,
+          selfie_image: image,
+        )
+
         expect(response.selfie_check_performed?).to be(true)
+        expect(response.extra).to have_key(:portrait_match_results)
       end
     end
 
     context 'when a liveness check is not required' do
       let(:liveness_checking_required) { false }
 
+      image = <<~YAML
+        doc_auth_result: Passed
+        failed_alerts: []
+      YAML
+
       it 'sets selfie_check_performed to false' do
+        response = client.post_images(
+          front_image: image,
+          back_image: image,
+          liveness_checking_required: liveness_checking_required,
+        )
+
         expect(response.selfie_check_performed?).to be(false)
+        expect(response.extra).not_to have_key(:portrait_match_results)
```

Comment:
This seems to me the best place to test that if we're not checking the selfie, we _don't_ have the `portrait_match_results` key, rather than building a whole new test for the negative condition. I thought making the opposite example of including the key above was then also warranted, and doing that changed the setup.